cmd/record.go

```diff
@@ -21,39 +23,75 @@ var recordCommand = &cobra.Command{
 	RunE:    record,
 }
 
-func record(cmd *cobra.Command, args []string) error {
-	if len(args) != 1 {
-		return fmt.Errorf("incorrect number of arguments provided. expected %d", 1)
+type recorder struct {
+	mu sync.Mutex
```

Comment:
nit: I think embedded mutexes are preferred unless they are publicly used.

Reply:
IIUC, embedding the mutex would expose public methods of mutex as part of recorder. I don't think we want this.

Reply:
IIUC, embedding the mutex would expose public methods of mutex as part of recorder. I don't think we want this.